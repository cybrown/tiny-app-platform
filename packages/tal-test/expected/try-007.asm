main()
  entry:
    Try               catchLabel: try_catch_1, endTryLabel: try_end_2
    Local             missing_variable
    TryPop
    Jump              try_end_2
  try_catch_1:
    ScopeEnter
    PushLatestError
    DeclareLocal      mutable: false, name: err, hasInitialValue: true
    Pop               inBlock: false
    Local             error
    ScopeLeave        inBlock: false
    Jump              try_end_2
  try_end_2:
