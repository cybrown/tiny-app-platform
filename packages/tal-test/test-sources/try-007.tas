missing_variable catch error
