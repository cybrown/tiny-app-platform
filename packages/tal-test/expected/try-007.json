[
  {
    "kind": "Try",
    "hasOnlyCatchKeyword": true,
    "node": {
      "kind": "Local",
      "name": "missing_variable"
    },
    "catchNode": {
      "kind": "Local",
      "name": "error"
    }
  }
]